### PR TITLE
19 classify with decision tree

### PR DESCRIPTION
**Pull request by Coop8**
Classification is done!

### CODE REVIEW

**Comment by Coop8** [comment]
Do not merge, we still need to do `determineDataCondition`

**Review by galaxybomb23 — APPROVED** [review]
looks good

File: include/DecisionTree.hpp [hunk 1]
```diff
@@ -27,7 +32,14 @@ struct BestFeatureResult {
 class DecisionTreeNode;
 class DecisionTree : public std::enable_shared_from_this<DecisionTree> {
   public:
+
+    // MARK: Discuss if these need to be private or can stay public
```
Comment (iBoot32) [inline include/DecisionTree.hpp:36]: should be completed before merge or after?

**Review by iBoot32 — COMMENTED** [review]
lgtm, just have my one comment above

File: include/DecisionTree.hpp (hunk 1 above)
Comment (Coop8) [reply include/DecisionTree.hpp:36]: prolly before, but since Eli merged we ball ig

**Pull request merged**